### PR TITLE
fixed: WS metrics

### PR DESCRIPTION
**Pull request by akhercha**
## Changed

- The metrics are now correctly recorded

### CODE REVIEW

File: pragma-node/src/types/ws/mod.rs [hunk 1]
```diff
@@ -180,11 +169,57 @@ where
...
+                        metrics::record_ws_interaction(
+                            Interaction::CloseConnection,
+                            Status::Success,
+                        );
+                        return Err(e);
+                    }
+                }
+            }
+        } else {
+            metrics::record_ws_interaction(Interaction::ClientMessageDecode, Status::Error);
```
[the remark this answers is not in the document]
Comment (akhercha) [reply pragma-node/src/types/ws/mod.rs:210]: i'd say no, returning an error here will propagate it to the WS endpoint & close the ws channel, we just want to warn the user with an error message but keep the channel alive

File: pragma-node/src/types/ws/mod.rs [hunk 2]
```diff
@@ -123,21 +125,8 @@ where
...
-                                        metrics::record_ws_interaction(Interaction::ClientMessage, Status::Success);
-                                    },
-                                    Err(e) => {
-                                        metrics::record_ws_interaction(Interaction::ClientMessage, Status::Error);
-                                        metrics::record_ws_interaction(Interaction::CloseConnection, Status::Success);
-                                        return Err(e);
-                                    }
-                                }
-                            }
+                            tracing::info!("👤 [CLIENT -> SERVER]");
```
[the remark this answers is not in the document]
Comment (akhercha) [reply pragma-node/src/types/ws/mod.rs:128]: done!

**Review by EvolveArt — APPROVED** [review]
LGTM

File: pragma-node/src/types/ws/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (akhercha) [reply pragma-node/src/types/ws/mod.rs:212]: but maybe we should if there's too many consecutives errors... like ~5 times spamming a wrong message should close the channel. shall I add that? @EvolveArt

[the remark this answers is not in the document]
Comment (EvolveArt) [reply pragma-node/src/types/ws/mod.rs:212]: Not necessary no, could lead to unexpected behavior

**Pull request merged**